en/_posts/2016-07-04-local-stateful.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title:  "Local stateful"
+categories: docker
+author: tgermain
+lang: en
+---
+
+
+
+This tutorial explains how local stateful and volumes work.
+
+
+
+# Volumes :
+
+The current version of volumes works exactly as docker host volume : the idea is to share a directory or a file between the host and the container.
+
+Example :
+
+```
+host $  mkdir -p ~/data/toto
+host $  touch ~/data/toto/hello
+host $  docker run -ti -v ~/data/toto:/data/ alpine sh
+container $ ls /data
+hello
+container $ touch /data/fromTheOtherSide
+host $ ls ~/data/toto/
+fromTheOtherSide  hello
+```
+
+Here we create a directory `~/data/toto` on the host, let's call it the **hostPath**. We create a file *hello* in it.
+
+When we run the container, we tell docker to map `~/data/toto` aka the **hostPath** to `/data/`, the **containerPath**, inside the container.
+
+The binding is bidirectionnal : if the container creates a file `/data/fromTheOtherSide`, we are able to see it from the host.
+
+
+
+To protect our infrastructure, we restrain what a user can mount into its containers.
```

Comment:
I'd rephrase it like, "There are some limitations on what you can mount in your container..." This puts the focus on the user rather than the technical/security constraints of the infrastructure.